pulp_deb/app/tasks/synchronizing.py

```diff
@@ -553,7 +573,32 @@ async def _handle_distribution(self, distribution):
         release_file = await self._create_unit(release_file_dc)
         if release_file is None:
             return
-        # Create release object
+        if self.optimize and distribution in self.repository.last_sync_details:
+            last_sync_dist_details = self.repository.last_sync_details[distribution]
+            if not self._has_remote_changed():
+                if (
+                    last_sync_dist_details["artifact_set_sha256"]
+                    == release_file.artifact_set_sha256
+                ):
```

Comment:
The entire condition could go in a single level as follows:
```suggestion
        if (
            self.optimize
            and distribution in self.repository.last_sync_details
            and not self._has_remote_changed()
            and self.repository.last_sync_details[distribution]["artifact_set_sha256"]
            == release_file.artifact_set_sha256
        ):
```